packages/component/src/Attachment/FileContent.js

```diff
@@ -0,0 +1,106 @@
+import { css } from 'glamor';
+import { format } from 'bytes';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import DownloadIcon from './Assets/DownloadIcon';
+import ScreenReaderText from '../ScreenReaderText';
+import useLocalize from '../hooks/useLocalize';
+import useStyleSet from '../hooks/useStyleSet';
+
+const ROOT_CSS = css({
+  display: 'flex',
+
+  '& .webchat__fileContent__buttonLink': {
+    display: 'flex',
+    flex: 1
+  },
+
+  '& .webchat__fileContent__badge': {
+    display: 'flex',
+    flex: 1,
+    flexDirection: 'column'
+  }
+});
+
+const FileContentBadge = ({ downloadIcon, fileName, size }) => {
+  const formattedSize = typeof size === 'number' && format(size);
+
+  return (
+    <React.Fragment>
+      <div aria-hidden={true} className="webchat__fileContent__badge">
+        <div className="webchat__fileContent__fileName">{fileName}</div>
+        {!!formattedSize && <div className="webchat__fileContent__size">{formattedSize}</div>}
+      </div>
+      {downloadIcon && <DownloadIcon className="webchat__fileContent__downloadIcon" size={1.5} />}
+    </React.Fragment>
+  );
+};
+
+FileContentBadge.defaultProps = {
+  downloadIcon: false,
+  size: undefined
+};
+
+FileContentBadge.propTypes = {
+  downloadIcon: PropTypes.bool,
+  fileName: PropTypes.string.isRequired,
+  size: PropTypes.number
+};
+
+const FileContent = ({ className, href, fileName, size }) => {
+  const formattedSize = format(size);
+
+  const [{ fileContent: fileContentStyleSet }] = useStyleSet();
+
+  const downloadLabel = useLocalize('Download file');
+  const uploadLabel = useLocalize('Upload file');
+  const downloadFileWithFileSizeLabel = useLocalize('DownloadFileWithFileSize', downloadLabel, fileName, formattedSize);
+  const uploadFileWithFileSizeLabel = useLocalize('UploadFileWithFileSize', uploadLabel, fileName, formattedSize);
+
+  const alt = href ? downloadFileWithFileSizeLabel : uploadFileWithFileSizeLabel;
+
+  return (
```

Comment:
Is there some accessibility reason why we have to use a ternary here? Seems like this can be simplified IMO.

```jsx
<div
  aria-hidden={true}
  className={classNames('webchat__fileContent', ROOT_CSS + '', fileContentStyleSet + '', (className || '') + '')}
>
  <ScreenReaderText text={alt} />
  {href &&  <a
    aria-hidden={true}
    className="webchat__fileContent__buttonLink"
    href={href}
    rel="noopener noreferrer"
    target="_blank"
    />}
  <FileContentBadge downloadIcon={!!href} fileName={fileName} size={size} />
</div>
```

Reply:
Haha, you are right.